elf.go

```diff
@@ -237,28 +238,77 @@ func (ec *elfCode) loadMaps(mapSections map[int]*elf.Section) (map[string]*MapSp
 	return maps, nil
 }
 
-func getProgType(v string) ProgType {
+func getProgType(v string) (ProgType, AttachType) {
 	types := map[string]ProgType{
 		// From https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/samples/bpf/bpf_load.c?id=fb40c9ddd66b9c9bb811bbee125b3cb3ba1faee7#n60
```

Comment:
Can you update this to reference https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/tools/lib/bpf/libbpf.c#n3568 instead? That seems to be a better source.